src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -96,158 +119,216 @@ public void Stop()
             {
-                // If previous block does not match there might have
-                // been a reorg, check if the wallet is still on the main chain.
-                ChainedHeader inBestChain = this.chain.GetBlock(this.walletTip.HashBlock);
-                if (inBestChain == null)
+                Guard.NotNull(block, nameof(block));
+                this.logger.LogTrace("({0}:'{1}')", nameof(block), block.GetHash());
+
+                ChainedHeader newTip = this.chain.GetBlock(block.GetHash());
+                if (newTip == null)
                 {
-                    // The current wallet hash was not found on the main chain.
-                    // A reorg happened so bring the wallet back top the last known fork.
-                    ChainedHeader fork = this.walletTip;
+                    this.logger.LogTrace("(-)[NEW_TIP_REORG]");
+                    return;
+                }
+
+                // If the new block's previous hash is the same as the
+                // wallet hash then just pass the block to the manager.
+                if (block.Header.HashPrevBlock != this.walletTip.HashBlock)
+                {
+                    // If previous block does not match there might have
+                    // been a reorg, check if the wallet is still on the main chain.
+                    ChainedHeader inBestChain = this.chain.GetBlock(this.walletTip.HashBlock);
+                    if (inBestChain == null)
+                    {
+                        // The current wallet hash was not found on the main chain.
+                        // A reorg happened so bring the wallet back top the last known fork.
+                        ChainedHeader fork = this.walletTip;
 
-                    // We walk back the chained block object to find the fork.
-                    while (this.chain.GetBlock(fork.HashBlock) == null)
-                        fork = fork.Previous;
+                        // We walk back the chained block object to find the fork.
+                        while (this.chain.GetBlock(fork.HashBlock) == null)
+                            fork = fork.Previous;
 
-                    this.logger.LogInformation("Reorg detected, going back from '{0}' to '{1}'.", this.walletTip, fork);
+                        this.logger.LogInformation("Reorg detected, going back from '{0}' to '{1}'.", this.walletTip, fork);
 
-                    this.walletManager.RemoveBlocks(fork);
-                    this.walletTip = fork;
+                        this.walletManager.RemoveBlocks(fork);
+                        this.walletTip = fork;
 
-                    this.logger.LogTrace("Wallet tip set to '{0}'.", this.walletTip);
+                        this.logger.LogTrace("Wallet tip set to '{0}'.", this.walletTip);
+                    }
+
+                    // The new tip can be ahead or behind the wallet.
+                    // If the new tip is ahead we try to bring the wallet up to the new tip.
```

Comment:
Was the change form > to < intended ?

Reply:
changed to <=

Reply:
Hi @monsieurleberre,

When I test wallet syncing the logic changed from > to < by mistake.  

I have reverted the change and testing to make sure the wallet sync's in the correct manner.

Cheers,

Ferdeen